api/src/opentrons/protocol_api/core/module.py

```diff
@@ -130,6 +134,149 @@ def get_status(self) -> MagneticStatus:
         """Get the module's current magnet status."""
 
 
+class AbstractThermocyclerCore(AbstractModuleCore[LabwareCoreType]):
+    """Core control interface for an attached Thermocycler Module."""
+
+    @abstractmethod
+    def open_lid(self) -> str:
+        """Open the thermocycler's lid."""
+
+    @abstractmethod
+    def close_lid(self) -> str:
+        """Close the thermocycler's lid."""
+
+    @abstractmethod
+    def set_target_block_temperature(
+        self,
+        celsius: float,
+        hold_time_seconds: Optional[float] = None,
+        block_max_volume: Optional[float] = None,
+    ) -> None:
+        """Set the target temperature for the well block, in °C.
+
+        Valid operational range yet to be determined.
+
+        :param celsius: The target temperature, in °C.
+        :param hold_time_seconds: The number of seconds to hold, after reaching
+                                  ``temperature``, before proceeding to the
+                                  next command. If `not specified, the
+                                  Thermocycler will proceed to the next
+                                  command after ``temperature`` is reached.
+        :param block_max_volume: The maximum volume of any individual well
+                                 of the loaded labware. If not supplied,
+                                 the thermocycler will default to 25µL/well.
+
+        .. note:
+
+            If ``hold_time_seconds`` is not specified, the Thermocycler
+            will proceed to the next command after ``temperature`` is reached.
+        """
```

Comment:
Let's use Google-style docstrings for internal interfaces, like the cores. Eventually, I'd like us to move the public ones over, too. See #7051 for history and context on this decision, and [Napoleon docs](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#docstring-sections) for writing guide.

```suggestion
        """Set the target temperature for the well block, in °C.

        Note:
            If ``hold_time_seconds`` is not specified, the Thermocycler
            will proceed to the next command after ``temperature`` is reached.

        Args:
            celsius: The target temperature, in °C.
            hold_time_seconds: The number of seconds to hold, after reaching
                ``temperature``, before proceeding to the next command.
            block_max_volume: The maximum volume of any individual well
                of the loaded labware. If not supplied, the thermocycler
                will default to 25µL/well.
        """
```